core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -750,6 +762,23 @@ public void copyLogSegmentsToRemote(UnifiedLog log) throws InterruptedException
                                         isCancelled(), isLeader());
                                 return;
                             }
+
+                            copyQuotaManagerLock.lock();
+                            try {
+                                while (rlmCopyQuotaManager.isQuotaExceeded()) {
+                                    logger.debug("Quota exceeded for copying log segments, waiting for the quota to be available.");
+                                    try {
+                                        copyQuotaManagerLockCondition.await(quotaTimeout().toMillis(), TimeUnit.MILLISECONDS);
+                                    } catch (InterruptedException e) {
```

Comment:
We are catching the InterruptedException:

1. does it delays the shutdown? If yes, do we have to add a check whether the task is cancelled before copying the log segment?
2. Why the logger level is set to WARN? can it be reduced to INFO?

Reply:
Addressed this. Also added comments.